include/picongpu/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -309,15 +309,20 @@ namespace picongpu
                     parentIon[momentum_] -= electronMomentum;
 
                     /** ionization of the ion by reducing the number of bound electrons
-                     *  and reset of atomic state to ground state of new charge state
                      *
                      * @warning subtracting a float from a float can potentially
                      *          create a negative boundElectrons number for the ion,
                      *          see #1850 for details
                      */
+                    parentIon[boundElectrons_] -= float_X(1.);
+
+                    uint8_t const newNumberBoundElectrons
+                        = static_cast<uint8_t>(parentIon[boundElectrons_]) - static_cast<uint8_t>(1u);
```

Comment:
we should avoid readin the value `parentIon[boundElectrons_]` twice. IMO this change is also reducing the bound electrons twice.

Reply:
offline discussed with @BrianMarre I am correct that we ionize now twice.
He will introduce a new method to set the charge state where bound electrons and atomic physics ground states will be handled correctly.

Reply:
After the fix there should be no code in PIConGPU except in the new method where `parentIon[boundElectrons_] = ...` is called.